test/Test/Database/LSMTree/Internal/Snapshot/Codec/Golden.hs

```diff
@@ -0,0 +1,267 @@
+      --
+      -- These file types' bindings have the following infix annotations, respectively:
+      --   1. (Fs) for FsPath
+      --   2. (Hs) for "Haskell" path
+      --   3. (Au) for "Golden file" path
+      snapshotFsPath = mkFsPath [name] <.> "snapshot"
+      snapshotHsPath = fsToFilePath goldenDataMountPoint snapshotFsPath
+      snapshotAuPath = snapshotHsPath <> ".golden"
+
+      -- IO actions
+      runnerIO :: FS.HasFS IO HandleIO
+      runnerIO = ioHasFS goldenDataMountPoint
+      removeIfExists :: FsPath -> IO ()
+      removeIfExists fp =
+        FS.doesFileExist runnerIO fp >>= (`when` (FS.removeFile runnerIO fp))
+      outputAction :: IO ()
+      outputAction = do
+        -- Ensure that if the output file already exists, we remove it and
+        -- re-write out the serialized data. This ensures that there are no
+        -- false-positives, false-negatives, or irrelavent I/O exceptions.
+        removeIfExists snapshotFsPath
+        BSL.writeFile snapshotHsPath . toLazyByteString $ encode datum
+
+  in  Au.goldenVsFile name snapshotAuPath snapshotHsPath outputAction
+
+testCodecSnapshotLabel :: TestTree
+testCodecSnapshotLabel =
+  let assembler (tagA, valA) =
+        let (tagB, valB) = basicSnapshotTableType
+            (tagC, valC) = basicTableConfig
+            (_tagD, valD) = basicRunNumber
+            (tagE, valE) = basicSnapLevels
+        in  (fuseAnnotations [tagA, tagB, tagC, tagE ], SnapshotMetaData valA valB valC valD valE)
+  in  testCodecBuilder "SnapshotLabels" $ assembler <$> enumerateSnapshotLabel
+
+testCodecSnapshotTableType :: TestTree
+testCodecSnapshotTableType =
+  let assembler (tagB, valB) =
+        let (tagA, valA) = basicSnapshotLabel
+            (tagC, valC) = basicTableConfig
+            (_tagD, valD) = basicRunNumber
+            (tagE, valE) = basicSnapLevels
+        in  (fuseAnnotations [tagA, tagB, tagC, tagE ], SnapshotMetaData valA valB valC valD valE)
+  in  testCodecBuilder "SnapshotTables" $ assembler <$> enumerateSnapshotTableType
+
+testCodecTableConfig :: TestTree
+testCodecTableConfig =
+  let assembler (tagC, valC) =
+        let (tagA, valA) = basicSnapshotLabel
+            (tagB, valB) = basicSnapshotTableType
+            (_tagD, valD) = basicRunNumber
+            (tagE, valE) = basicSnapLevels
+        in  (fuseAnnotations [tagA, tagB, tagC, tagE ], SnapshotMetaData valA valB valC valD valE)
+  in  testCodecBuilder "SnapshotConfig" $ assembler <$> enumerateTableConfig
+
+testCodecSnapLevels :: TestTree
+testCodecSnapLevels =
+  let assembler (tagE, valE) =
+        let (tagA, valA) = basicSnapshotLabel
+            (tagB, valB) = basicSnapshotTableType
+            (tagC, valC) = basicTableConfig
+            (_tagD, valD) = basicRunNumber
+        in  (fuseAnnotations [tagA, tagB, tagC, tagE ], SnapshotMetaData valA valB valC valD valE)
+  in  testCodecBuilder "SnapshotLevels" $ assembler <$> enumerateSnapLevels
```

Comment:
Shouldn't `_tagD` be included in the name?

Reply:
Yeah, I cleaned this up.